src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -397,7 +397,14 @@ private static void CheckSnapshot(ITextSnapshot snapshot)
 
             private IEnumerable<ITagSpan<TTag>> GetNonIntersectingTagSpans(IEnumerable<SnapshotSpan> spansToInvalidate, TagSpanIntervalTree<TTag> oldTagTree)
             {
-                var snapshot = spansToInvalidate.First().Snapshot;
+                var firstSpanToInvalidate = spansToInvalidate.First();
+
+                // Performance: No need to fully realize spansToInvalidate or do any of the calculations below if the
+                //   full snapshot is being invalidated.
+                if (firstSpanToInvalidate.Length == firstSpanToInvalidate.Snapshot.Length)
+                    return Array.Empty<ITagSpan<TTag>>();
```

Comment:
fascinating.  do we know why the entire span is being invalidated?

Reply:
From what I can see, very few of the Roslyn taggers override [GetSpansToTag](https://github.com/dotnet/roslyn/blob/0caa880a161057559d1dcaad890e4053f520136e/src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.cs#L213)